Added Dockerfile and build instructions for Jetson. Also set CUDA arch set automatically. (#4637)
microsoft/onnxruntime: dockerfiles/README.md

- OpenVINO: [Dockerfile](Dockerfile.openvino), [Instructions](#openvino)
- Nuphar: [Dockerfile](Dockerfile.nuphar), [Instructions](#nuphar)
- ARM 32v7: [Dockerfile](Dockerfile.arm32v7), [Instructions](#arm-32v7)

> - NVIDIA Jetson TX1/TX2/Nano/N: [Dockerfile](Dockerfile.jetson), [Instructions](#nvidia-jetson-tx1tx2nanon)

Reviewer: should N be Xavier instead?

    ```
10. Test installation by following the instructions [here](https://microsoft.github.io/onnxruntime/)

> ## NVIDIA Jetson TX1/TX2/Nano/N:

Reviewer: Would Jetson TX1/TX2/Nano/Xavier be a more accurate description?

These instructions are for [JetPack SDK 4.4](https://developer.nvidia.com/embedded/jetpack).
The Dockerfile.jetson is using [NVIDIA L4T 32.4.3](https://developer.nvidia.com/embedded/linux-tegra) as base image.

> You can probably used those instructions with earlier/later versions with appropriate modifications.

Reviewer 2: ```suggestion
Versions different from these may require modifications to these instructions.
```

Author: Thanks for editorial comments! Will address them today.

> Instructions assume you are on Jetson host in the root of onnxruntime git project clone.

Reviewer 2: ```suggestion
These instructions assume you are on Jetson host in the root of onnxruntime git project clone (`https://github.com/microsoft/onnxruntime`).
```

Two-step installation is required:

1. Build Python 'wheel' for ONNXruntime on host Jetson system;
2. Build Docker image using ONNXruntime wheel from step 1. You can also install the wheel on the host directly.

Here are the build commands for each step:

> 1.1 Install ONNXruntime build dependencies on Jetpack 4.4 host:

Reviewer 2: ```suggestion
1.1 Install ONNX Runtime build dependencies on Jetpack 4.4 host:
```

```
   sudo apt install -y --no-install-recommends \
    	build-essential software-properties-common cmake libopenblas-dev \
	libpython3.6-dev python3-pip python3-dev
```

> 1.2 Build ONNXRuntime Python wheel:

Reviewer 2: ```suggestion
1.2 Build ONNX Runtime Python wheel:
```

```
   ./build.sh --update --config Release --build --build_wheel \

>    --use_cuda --cuda_home /usr/local/cuda --cudnn_home /usr/lib/aarch64-linux-gnu

Reviewer 3: >--use_cuda [](start = 3, length = 10)

Is this for CUDA EP only? the docker file should be able to support TensorRT also. why not make it another option to build TensorRT also?
Should also mention the TRT + CUDA + cudnn version combinations for Jetpack 4.4.

Reviewer: yes it's purposely for CUDA EP only. TRT 7.1 support is not yet officially integrated for ORT.

```
Note: You may add --use_tensorrt and --tensorrt_home options if you wish to use NVIDIA TensorRT (support is experimental), as well as any other options supported by [build.sh script](build.sh).

2. After the Python wheel is successfully built, use 'find' command for Docker to install the wheel inside new image:
```
   find . -name '*.whl' -print -exec sudo -H DOCKER_BUILDKIT=1 nvidia-docker build --build-arg WHEEL_FILE={} -f ./dockerfiles/Dockerfile.jetson . \;
```

> Note: Resulting Docker image will have ONNXruntime installed in /usr, and ONNXruntime wheel copied to /onnxruntime directory.

Reviewer 2: ```suggestion
Note: Resulting Docker image will have ONNX Runtime installed in /usr, and ONNX Runtime wheel copied to /onnxruntime directory.
```

> Nothing else from ONNXruntime source tree will be copied/installed to the image.

Reviewer 2: ```suggestion
Nothing else from ONNX Runtime source tree will be copied/installed to the image.
```
